testclock/clock.go

```diff
@@ -212,6 +259,32 @@ func (clock *Clock) reset(t *timer, d time.Duration) bool {
 	return found
 }
 
+// reset is the underlying implementation of clock.Alarm.Reset, which may be
+// called by any Alarm backed by this Clock.
+func (clock *Clock) resetTime(t *timer, deadline time.Time) bool {
+	defer clock.notifyAlarm()
+	clock.mu.Lock()
+	defer clock.mu.Unlock()
+
+	found := false
+	for _, wt := range clock.waiting {
+		if wt == t {
+			found = true
+		}
+	}
+	if !found {
+		clock.waiting = append(clock.waiting, t)
+	}
+	t.deadline = deadline
+	sort.Sort(byDeadline(clock.waiting))
+	if clock.now.After(t.deadline) {
+		// If the time has already passed, that means we should be triggering the
+		// Timer right away, as "now" has already occured.
```

Comment:
**todo(typo):** `occurred` (seams to be a cc i think there is more than one with `occur` verb